packages/deployments/contracts/contracts/nomad-xapps/contracts/connext/ConnextHandler.sol

```diff
@@ -0,0 +1,530 @@
+    }
+
+    ConnextUtils.ExecuteLibArgs memory libArgs = ConnextUtils.ExecuteLibArgs({
+      executeArgs: _args,
+      maxRoutersPerTransfer: maxRoutersPerTransfer,
+      tokenRegistry: tokenRegistry,
+      wrapper: wrapper,
+      executor: executor,
+      LIQUIDITY_FEE_NUMERATOR: LIQUIDITY_FEE_NUMERATOR,
+      LIQUIDITY_FEE_DENOMINATOR: LIQUIDITY_FEE_DENOMINATOR
+    });
+
+    return
+      ConnextUtils.execute(
+        libArgs,
+        routedTransfers,
+        reconciledTransfers,
+        routerBalances,
+        adoptedToLocalPools,
+        canonicalToAdopted,
+        routerInfo,
+        transferRelayer
+      );
+  }
+
+  /**
+   * @notice Anyone can call this function on the origin domain to increase the relayer fee for a transfer.
+   * @param _transferId - The unique identifier of the crosschain transaction
+   */
+  function bumpTransfer(bytes32 _transferId) external payable {
+    ConnextUtils.bumpTransfer(_transferId, relayerFees);
+  }
+
+  /**
+   * @notice Called by relayer when they want to claim owed funds on a given domain
+   * @dev Domain should be the origin domain of all the transfer ids
+   * @param _recipient - address on origin chain to send claimed funds to
+   * @param _domain - domain to claim funds on
+   * @param _transferIds - transferIds to claim
+   */
+  function initiateClaim(
+    uint32 _domain,
+    address _recipient,
+    bytes32[] calldata _transferIds
+  ) external override {
+    ConnextUtils.initiateClaim(_domain, _recipient, _transferIds, relayerFeeRouter, transferRelayer);
+  }
+
+  /**
+   * @notice Pays out a relayer for the given fees
+   * @dev Called by the RelayerFeeRouter.handle message. The validity of the transferIds is
+   * asserted before dispatching the message.
+   * @param _recipient - address on origin chain to send claimed funds to
+   * @param _transferIds - transferIds to claim
+   */
+  function claim(address _recipient, bytes32[] calldata _transferIds) external override onlyRelayerFeeRouter {
+    ConnextUtils.claim(_recipient, _transferIds, relayerFees);
+  }
+
+  // ============ Internal functions ============
+
+  /**
+   * @notice Contains the logic to verify + increment a given routers liquidity
+   * @dev The liquidity will be held in the local asset, which is the representation if you
+   * are *not* on the canonical domain, and the canonical asset otherwise.
+   * @param _amount - The amount of liquidity to add for the router
+   * @param _local - The address of the nomad representation of the asset
+   * @param _router - The router you are adding liquidity on behalf of
+   */
+  function _addLiquidityForRouter(
+    uint256 _amount,
+    address _local,
+    address _router
+  ) internal {
+    // Sanity check: router is sensible
+    if (_router == address(0)) revert ConnextHandler__addLiquidityForRouter_routerEmpty();
+
+    // Sanity check: nonzero amounts
+    if (_amount == 0) revert ConnextHandler__addLiquidityForRouter_amountIsZero();
```

Comment:
Note that amount will be 0 if the router is adding liquidity for native assets... no?

Reply:
ik we dont support that yet, so i figured maybe that's why this is here. but just double checking

Reply:
Amount will never be 0, even is using native assets. It should always be the amount you add, and is asserted in the `AssetLogic` library